Modules/md5module.c

```diff
@@ -279,7 +300,13 @@ _md5_md5_impl(PyObject *module, PyObject *string, int usedforsecurity)
         return NULL;
     }
     if (string) {
-        update(new->hash_state, buf.buf, buf.len);
+        if (buf.len >= HASHLIB_GIL_MINSIZE) {
```

Comment:
I presume this does not rely on a lock because by definition, no one else is aware of this newly-created object, yet?

Reply:
exactly!